src/fparser/common/splitline.py

```diff
@@ -249,94 +250,58 @@ def splitquote(line, stopchar=None, lower=False, quotechars="\"'"):
-                    ipos -= 1
-                    # This marks the end of the current part.
-                    break
-                if char == "\\":
-                    nofslashes += 1
+    segments = []
+    i = 0
+    n = len(line)
+    quote_char = stopchar
+
+    while i < n:
+        if quote_char is None and line[i] in ("'", '"'):
+            quote_char = line[i]
+            start = i
+            i += 1
+            while i < n:
+                if line[i] == quote_char:
+                    if i + 1 < n and line[i + 1] == quote_char:
+                        i += 2  # Escaped quote
+                    else:
+                        i += 1
+                        break
                 else:
-                    nofslashes = 0
-                part.append(char)
-                try:
-                    char = line[ipos]
-                    ipos += 1
-                except IndexError:
-                    break
-            if part:
-                # Found a part. Add it to the list of items.
-                item = "".join(part)
-                if lower:
-                    item = item.lower()
-                items.append(item)
-            # Move on to the next character in the line.
-            continue
-        if char == stopchar:
-            # string starts with quotechar
-            part.append(char)
-            try:
-                char = line[ipos]
-                ipos += 1
-            except IndexError:
-                # Have reached the end of the line after encountering an
-                # opening quote character.
-                if part:
-                    item = String("".join(part))
-                    items.append(item)
-                break
-        # else continued string
-        while 1:
-            if char == stopchar and not nofslashes % 2:
-                # We've found the closing quote character.
-                part.append(char)
-                stopchar = None
-                break
-            if char == "\\":
-                nofslashes += 1
+                    i += 1
+            if i > n or (i == n and line[i - 1] != quote_char):
+                segment = String(line[start:])
+                segments.append(segment)
+                return [
+                    s if isinstance(s, String) else s.lower() if lower else s
+                    for s in segments
+                ], quote_char
             else:
-                nofslashes = 0
-            part.append(char)
-            try:
-                char = line[ipos]
-                ipos += 1
-            except IndexError:
-                break
-        if part:
-            item = String("".join(part))
-            items.append(item)
-    return items, stopchar
+                segment = String(line[start:i])
+                segments.append(segment)
+                quote_char = None
+        else:
+            start = i
+            while i < n and (quote_char is not None or line[i] not in ("'", '"')):
```

Comment:
I am not really convinced that this is correct (and if I am wrong, it needs some comments):
if the caller sets `stop_char` (which is what `quote_char` is initialised with), then I believe this while loop becomes basically:
```
while i < n and True:
```
i.e. it will just return the rest of the string, ignoring any potential `stop_char` in the string?? I also didn't see a test that provides the `stop_char` to the call.

Reply:
Darn, you are right. So much for AI.

Reply:
Human: 1, AI: 0